### PR TITLE
validate category name

### PR DESCRIPTION
**Pull request by ghost**
## Description
validate category name when update to database
## Reference Requirements
- Product Backlog ID: PBI-7
- Task ID: 7.1, 7.2

## Reference Isuees
#{issue_id}

## Migrations
YES
## Impacted Areas in Application
List general components of the application that this PR will affect:
No

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)
- query
update `categories` set `name` = 'Dr. Kelvin Leuschke23', `updated_at` = '2018-02-01 02:46:37' where `id` = '1'

- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)
image

### CODE REVIEW

File: resources/views/backend/categories/index.blade.php [hunk 1]
```diff
@@ -58,15 +58,20 @@
                       <div class="form-group">
                         <label class="control-label col-sm-2" for="name">{{ __('category.name') }}:</label>
                         <div class="col-sm-10">
-                          <input type="name" class="form-control" id="nameCategory" autofocus>
+                          <input type="text" class="form-control" id="nameCategory" autofocus>
```
Comment (at-luanle) [inline resources/views/backend/categories/index.blade.php]: lỗi convention vd: nameCategory = name-category

File: resources/views/backend/categories/index.blade.php [hunk 2]
```diff
@@ -58,15 +58,20 @@
...
+                      </div>
+                      <div class="form-group">
+                        <div class="col-sm-10 col-sm-push-2">
+                          <p class="errors text-warning"></p>
                         </div>
                       </div>
                     </form>
                     <div class="modal-footer">
-                      <button type="button" class="btn btn-success btn-UpdateNameCategory" data-dismiss="modal">
+                      <button type="button" class="btn btn-success btn-UpdateNameCategory">
```
Comment (at-luanle) [inline resources/views/backend/categories/index.blade.php]: lỗi convention : btn-UpdateNameCategory = btn-update-name-category

File: resources/views/backend/categories/load-category-list.blade.php [hunk 3]
```diff
@@ -37,15 +37,20 @@
           <div class="form-group">
             <label class="control-label col-sm-2" for="name">{{ __('category.name') }}:</label>
             <div class="col-sm-10">
-              <input type="name" class="form-control" id="nameCategory" autofocus>
+              <input type="text" class="form-control" id="nameCategory" autofocus>
```
Comment (at-luanle) [inline resources/views/backend/categories/load-category-list.blade.php]: lỗi convention : nameCategory = name-category

File: resources/views/backend/categories/load-category-list.blade.php [hunk 4]
```diff
@@ -37,15 +37,20 @@
...
+          </div>
+          <div class="form-group">
+            <div class="col-sm-10 col-sm-push-2">
+              <p class="errors text-warning"></p>
             </div>
           </div>
         </form>
         <div class="modal-footer">
-          <button type="button" class="btn btn-success btn-UpdateNameCategory" data-dismiss="modal">
+          <button type="button" class="btn btn-success btn-UpdateNameCategory">
```
Comment (at-luanle) [inline resources/views/backend/categories/load-category-list.blade.php]: lỗi convention :  UpdateNameCategory = btn-update-name-category

File: resources/views/backend/categories/index.blade.php (hunk 1 above)
Comment (at-nghiavo) [reply resources/views/backend/categories/index.blade.php]: name do task trước đặt, em sửa lại type thôi a.

File: resources/views/backend/categories/load-category-list.blade.php (hunk 3 above)
Comment (at-nghiavo) [reply resources/views/backend/categories/load-category-list.blade.php]: em sửa type thôi

File: resources/views/backend/categories/load-category-list.blade.php (hunk 4 above)
Comment (at-nghiavo) [reply resources/views/backend/categories/load-category-list.blade.php]: em remove att data-dismiss

**Comment by at-luanle** [comment]
LGTM

File: resources/views/backend/categories/index.blade.php (hunk 1 above)
Comment (at-xinhnguyen) [reply resources/views/backend/categories/index.blade.php]: thấy code cũ sai thì nên sửa luôn đi Nghĩa

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**